### PR TITLE
configure: add flag to disable OpenSSL tests

### PR DESCRIPTION
**Pull request by apoelstra**
OpenSSL messes up valgrind.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -323,16 +328,25 @@ esac
...
+
+      case $host in
+      *mingw*)
+        SECP_TEST_LIBS="$SECP_TEST_LIBS -lgdi32"
+        ;;
+      esac
+    fi
+  else
+    if test x"$enable_openssl_tests" = x"yes"; then
+      AC_MSG_ERROR([OpenSSL tests requested but not OpenSSL with EC support is not available])
```
Comment (sipa) [inline configure.ac]: Not not?

**Comment by sipa** [comment]
```
src/tests.c: In function ‘test_ecdsa_der_parse’:
src/tests.c:3503:32: warning: unused variable ‘max_scalar’ [-Wunused-variable]
     static const unsigned char max_scalar[32] = {
```

**Comment by afk11** [comment]
See also: https://github.com/bitcoin-core/secp256k1/pull/382

**Comment by apoelstra** [comment]
Thanks @afk11, fixed.

**Comment by sipa** [comment]
ACK

**Pull request merged**